sdk/iot/hub/tests/cmocka/test_az_iot_hub_client.c

```diff
@@ -89,28 +109,16 @@ static void test_az_iot_hub_client_user_name_get_succeed(void** state)
   assert_int_equal(
       az_iot_hub_client_init(&client, test_device_hostname, test_device_id, NULL), AZ_OK);
 
-  uint8_t test_span_buffer[sizeof(test_correct_user_name) - 1];
-  az_span test_span = az_span_init(test_span_buffer, 0, _az_COUNTOF(test_span_buffer));
+  uint8_t test_span_buf[TEST_SPAN_BUFFER_SIZE];
+  memset(test_span_buf, 0xFF, _az_COUNTOF(test_span_buf));
```

Comment:
You can also `az_span_set` if you have that buffer as a span.

Reply:
yea i thought about that. this was honestly easier to copy and paste once i got going haha

Reply:
I like your offline idea of extracting a method and reuse it everywhere. Maybe we can add another assert that's SDK-specific to the az_core/test folder? (Similar to preconditions?)
assert_az_span_equal(a, b). It could also have proper string output to simplify diffs.